src/sparktuner/spark_logs.py

```diff
@@ -1,10 +1,56 @@
+import re
+import logging
+
+log = logging.getLogger(__name__)
+
+
 class SparkLogProcessor(object):
     """
-    Parses Spark application logs
+    Processes Spark logs
     """
-    def parse_yarn_app_log(self):
+    def __init__(self, master, deploy_mode):
+        self.yarn_app_id = ""
```

Comment:
It seems a little weird/not intuitive that the yarn_app_id would be defined in a class called "SparkLogProcessor" but at the moment, I don't have a good idea of how to improve the structure, other than maybe returning the app id at some point?

Reply:
It does since it seems like an unnecessary dependency between log parsing and application behavior. I'll think about better ways